jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/ActionSleep.java

```diff
@@ -50,45 +52,57 @@ public void validate(HttpAction action) {}
 
     @Override
     public void execute(HttpAction action) {
-        Runnable task = createRunnable(action);
+        SleepTask task = createRunnable(action);
         AsyncTask aTask = Async.execASyncTask(action, AsyncPool.get(), "sleep", task);
+        action.log.info(format("[%d] Sleep %d ms.", action.id, task.sleepMilli));
         JsonValue v = Async.asJson(aTask);
-        Async.setLocationHeader(action, aTask);
         ServletOps.sendJsonReponse(action, v);
     }
 
-    protected Runnable createRunnable(HttpAction action) {
+    protected SleepTask createRunnable(HttpAction action) {
         String interval = action.request.getParameter("interval");
         int sleepMilli = 5000;
-        if ( interval != null )
+        if ( interval != null ) {
             try {
                 sleepMilli = Integer.parseInt(interval);
             } catch (NumberFormatException ex) {
-                action.log.error(format("[%d] NumberFormatException: %s", action.id, interval));
+                ServletOps.errorBadRequest("Bad format for 'interval': integer required");
```

Comment:
Maybe use the same prefix for all errors?

`"Bad format for 'interval': integer required"`

`"Bad format for 'interval': negative sleep interval"`

`"Bad format for 'interval': sleep internal greater than maximum allowed"`

?

Reply:
Good idea.